components/analytics/true_up_review.tsx

```diff
@@ -0,0 +1,230 @@
+            className={classNames('btn btn-primary TrueUpReview__submit', {'TrueUpReview__submit--error': trueUpReviewError})}
+            onClick={isAirGapped ? handleDownloadBundle : handleSubmitReview}
+        >
+            {isAirGapped ?
+                <FormattedMessage
+                    id='admin.billing.trueUpReview.button_download'
+                    defaultMessage='Download Data'
+                /> :
+                <FormattedMessage
+                    id='admin.billing.trueUpReview.button_share'
+                    defaultMessage='Share to Mattermost'
+                />
+            }
+        </button>
+    );
+
+    const errorStatus = (
+        <>
+            <WarningIcon additionalClassName={'TrueUpReview__warning'}/>
+            <FormattedMessage
+                id='admin.billing.trueUpReview.submit_error'
+                defaultMessage='There was an issue sending your True Up Review. Please try again.'
+            />
+            {submitButton}
+        </>
+    );
+
+    const successStatus = (
+        <>
+            <CheckMarkSvg/>
+            <FormattedMessage
+                id='admin.billing.trueUpReview.submit_success'
+                defaultMessage='Success!'
+            />
+            <FormattedMessage
+                id='admin.billing.trueUpReview.submit.thanks_for_sharing'
+                defaultMessage='Thanks for sharing data needed for your true-up review.'
+            />
+        </>
+    );
+
+    const reviewDetails = (
+        <>
+            {dueDate}
+            <FormattedMessage
+                id='admin.billing.trueUpReview.share_data_for_review'
+                defaultMessage='Share the below workspace data with Mattermost for your quarterly true-up Review.'
+            />
+            {submitButton}
+        </>
+    );
+
+    const cardContent = () => {
+        if (reviewProfile.getRequestState !== 'OK' && trueUpReviewError) {
+            return errorStatus;
+        }
+
+        // If we just submitted and the review status is set as complete, show the success
+        // status details.
+        if (reviewProfile.getRequestState === 'OK') {
+            return successStatus;
+        }
+
+        // If the due date is empty we still have the default state.
+        if (!reviewStatus.due_date) {
+            return null;
+        }
+
+        return reviewDetails;
+    };
+
+    // Only show the true up review section if the user is an admin and we're not using a cloud instance.
+    if (!licenseIsTrueUpEligible || !isSystemAdmin) {
+        return null;
+    }
+
+    // Only display the review details if we are within 2 weeks of the review due date.
+    const visibilityStart = moment(reviewStatus.due_date).startOf('day').subtract(2, 'weeks');
+    if (moment().isSameOrBefore(visibilityStart)) {
+        return null;
+    }
+
+    // If the review has already been submitted, don't show anything.
+    if (reviewStatus.complete) {
+        return null;
+    }
```

Comment:
Should we also check here if it's past the due date so it doesn't show after that even if they didn't submit it?

Reply:
Yes, this is covered in the condition above this one (lines 192 - 195).
The panel will only be visible if we're within the 2-week window.